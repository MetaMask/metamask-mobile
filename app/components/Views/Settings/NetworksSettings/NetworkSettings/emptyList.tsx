import React from 'react';
import { StyleSheet, Text } from 'react-native';
import { useNavigation } from '@react-navigation/native';
import { strings } from '../../../../../../locales/i18n';
import Alert, { AlertType } from '../../../../Base/Alert';
import { useTheme } from '../../../../../util/theme';
import { CHAINLIST_URL } from '../../../../../constants/urls';
import Routes from '../../../../../constants/navigation/Routes';

const createStyles = (colors: any) =>
  StyleSheet.create({
    wrapper: {
      backgroundColor: colors.background.default,
      flex: 1,
    },
    container: { marginHorizontal: 10, marginTop: 20, paddingRight: 0 },
    emptyDescriptionText: { color: colors.text.default },
    link: { color: colors.primary.default },
  });

interface Props {
  goToCustomNetwork: () => void;
}

const EmptyPopularList = ({ goToCustomNetwork }: Props) => {
  const { colors } = useTheme();
  const styles = createStyles(colors);
  const navigation = useNavigation();

  const goToBrowserTab = () => {
    navigation.navigate('BrowserTabHome', {
      screen: Routes.BROWSER.VIEW,
      params: {
        newTabUrl: CHAINLIST_URL,
        timestamp: Date.now(),
      },
    });
  };

  return (
    <Alert type={AlertType.Info} style={styles.container}>
      <>
        <Text style={styles.emptyDescriptionText}>{`${strings(
          'networks.empty_popular_networks',
        )} `}</Text>
        <Text style={styles.link} onPress={goToBrowserTab}>{`${strings(
          'networks.add_other_network_here',
        )} `}</Text>
        <Text style={styles.emptyDescriptionText}>{`${strings(
          'networks.you_can',
        )} `}</Text>
        <Text
          suppressHighlighting
          onPress={goToCustomNetwork}
          style={styles.link}
        >
          {strings('networks.add_network')}
        </Text>
      </>
    </Alert>
  );
};

export default EmptyPopularList;
