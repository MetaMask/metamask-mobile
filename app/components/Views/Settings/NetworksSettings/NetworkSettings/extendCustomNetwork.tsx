import React from 'react';
import NetworkModals from '../../../../UI/NetworkModal';
import { View, TouchableOpacity } from 'react-native';
import ImageIcons from '../../../../UI/ImageIcon';
import WarningIcon from 'react-native-vector-icons/FontAwesome';
import CustomText from '../../../../Base/Text';
import EmptyPopularList from './emptyList';
import { useNavigation } from '@react-navigation/native';
import { strings } from '../../../../../../locales/i18n';
import { useTheme } from '../../../../../util/theme';
import PopularList from '../../../../../util/networks/customNetworks';
import createStyles from './styles';

const ExtendedNetworkList = ({
  frequentRpcList,
  showPopularNetworkModal,
  onCancel,
  popularNetwork,
  toggleWarningModal,
  togglePopularNetwork,
  tabView,
}: any) => {
  // The frequentRpcList is a list of network the user has already added.
  // The popularNetwork list should hide this list.
  // However, the on-ramp will include this list to the defined popular list but change the CTA to "Switch"
  const navigation = useNavigation();
  const { colors } = useTheme();
  const styles = createStyles(colors);
  const filteredPopularList = PopularList.filter(
    (val) =>
      !frequentRpcList.some(
        (key: { chainId: string }) => val.chainId === key.chainId,
      ),
  );

  // this condition won't matter for the on ramp.
  if (filteredPopularList.length === 0) {
    return <EmptyPopularList goToCustomNetwork={() => tabView.goToPage(1)} />;
  }

  return (
    <>
      {showPopularNetworkModal && (
        <NetworkModals
          isVisible={showPopularNetworkModal}
          onClose={onCancel}
          network={popularNetwork}
          navigation={navigation}
          listOfAddedNetworks={filteredPopularList}
        />
      )}
      {filteredPopularList.map((item, index) => (
        <TouchableOpacity
          key={index}
          style={styles.popularNetwork}
          onPress={() => togglePopularNetwork(item)}
        >
          <View style={styles.popularWrapper}>
            <ImageIcons
              image={item.rpcPrefs.imageUrl}
              style={styles.popularNetworkImage}
            />
            <CustomText bold>{item.nickname}</CustomText>
          </View>
          <View style={styles.popularWrapper}>
            {item.warning ? (
              <WarningIcon
                name="warning"
                size={14}
                color={colors.icon.alternative}
                style={styles.icon}
                onPress={toggleWarningModal}
              />
            ) : null}
            <CustomText link>{strings('networks.add')}</CustomText>
          </View>
        </TouchableOpacity>
      ))}
    </>
  );
};

export default ExtendedNetworkList;