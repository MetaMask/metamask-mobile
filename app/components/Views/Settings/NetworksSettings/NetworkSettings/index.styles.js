import { typography } from '@metamask/design-tokens';
import { StyleSheet } from 'react-native';
import Device from '../../../../../util/device';
import {
  fontStyles,
  colors as staticColors,
} from '../../../../../styles/common';
import {
  getFontFamily,
  TextVariant,
} from '../../../../../component-library/components/Texts/Text';

export const createStyles = (colors) =>
  StyleSheet.create({
    baseAll: {
      padding: 16,
    },
    container: {
      flex: 1,
    },
    scrollViewContent: {
      height: '100%',
    },
    scrollableBox: {
      marginTop: 24,
      justifyContent: 'center',
      alignItems: 'center',
      alignSelf: 'center',
    },
    addRpcNameButton: {
      paddingTop: 32,
      alignSelf: 'center',
    },
    sheet: {
      flexDirection: 'column',
      bottom: 0,
      top: Device.getDeviceHeight() * 0.5,
      backgroundColor: colors.background.default,
      borderTopLeftRadius: 10,
      borderTopRightRadius: 10,
      height: Device.getDeviceHeight() * 0.5,
    },
    sheetSmall: {
      position: 'absolute',
      bottom: 0,
      top: Device.getDeviceHeight() * 0.7,
      backgroundColor: colors.background.default,
      borderTopLeftRadius: 10,
      borderTopRightRadius: 10,
      height: Device.getDeviceHeight() * 0.3,
    },
    sheetRpcForm: {
      position: 'absolute',
      bottom: 0,
      top: Device.getDeviceHeight() * 0.3,
      backgroundColor: colors.background.default,
      borderTopLeftRadius: 10,
      borderTopRightRadius: 10,
      height: '100%',
    },
    notch: {
      width: 48,
      height: 5,
      borderRadius: 4,
      backgroundColor: colors.border.default,
      marginTop: 4,
      alignSelf: 'center',
    },
    rpcMenu: {
      paddingHorizontal: 16,
      flex: 1,
    },
    wrapper: {
      backgroundColor: colors.background.default,
      flexGrow: 1,
      flexDirection: 'column',
    },
    informationWrapper: {
      flex: 1,
      paddingHorizontal: 16,
    },
    informationCustomWrapper: {},
    scrollWrapper: {
      flex: 1,
      paddingVertical: 12,
    },
    scrollWrapperOverlay: {
      flex: 1,
      paddingVertical: 12,
      opacity: 0.5,
    },
    onboardingInput: {
      borderColor: staticColors.transparent,
      padding: 0,
    },
    inputDisabled: {
      borderColor: colors.border.muted,
      color: colors.text.muted,
    },
    input: {
      ...fontStyles.normal,
      borderColor: colors.border.default,
      borderRadius: 5,
      borderWidth: 2,
      padding: 10,
      color: colors.text.default,
    },
    dropDownInput: {
      borderColor: colors.border.default,
      borderRadius: 5,
      borderWidth: 2,
    },
    inputWithError: {
      ...typography.sBodyMD,
      fontFamily: getFontFamily(TextVariant.BodyMD),
      borderColor: colors.error.default,
      borderRadius: 5,
      borderWidth: 1,
      paddingTop: 2,
      paddingBottom: 12,
      paddingHorizontal: 12,
      color: colors.text.default,
    },
    inputWithFocus: {
      ...typography.sBodyMD,
      fontFamily: getFontFamily(TextVariant.BodyMD),
      borderColor: colors.primary.default,
      borderRadius: 5,
      borderWidth: 2,
      paddingTop: 2,
      paddingBottom: 12,
      paddingHorizontal: 12,
      color: colors.text.default,
    },
    warningText: {
      ...fontStyles.normal,
      color: colors.error.default,
      marginTop: 4,
      paddingLeft: 2,
      paddingRight: 4,
    },
    warningContainer: {
      marginTop: 16,
      flexGrow: 1,
      flexShrink: 1,
    },
    newWarningContainer: {
      flexGrow: 1,
      flexShrink: 1,
    },
    heading: {
      fontSize: 16,
      color: colors.text.default,
      ...fontStyles.bold,
    },
    label: {
      fontSize: 14,
      paddingVertical: 12,
      color: colors.text.default,
      ...fontStyles.bold,
    },
    link: {
      color: colors.primary.default,
    },
    title: {
      fontSize: 20,
      paddingVertical: 12,
      color: colors.text.default,
      ...fontStyles.bold,
    },
    desc: {
      fontSize: 14,
      color: colors.text.default,
      ...fontStyles.normal,
    },
    messageWarning: {
      paddingVertical: 2,
      fontSize: 14,
      color: colors.warning.default,
      ...typography.sBodyMD,
      fontFamily: getFontFamily(TextVariant.BodyMD),
    },
    inlineWarning: {
      paddingVertical: 2,
      fontSize: 14,
      color: colors.text.default,
      ...typography.sBodyMD,
      fontFamily: getFontFamily(TextVariant.BodyMD),
    },
    inlineWarningMessage: {
      paddingVertical: 2,
      color: colors.warning.default,
      ...typography.sBodyMD,
      fontFamily: getFontFamily(TextVariant.BodyMD),
    },
    buttonsWrapper: {
      marginVertical: 12,
      flexDirection: 'row',
      alignSelf: 'flex-end',
    },
    buttonsContainer: {
      flex: 1,
      flexDirection: 'column',
      alignSelf: 'flex-end',
    },
    editableButtonsContainer: {
      flex: 1,
      flexDirection: 'row',
    },
    networksWrapper: {
      marginTop: 12,
    },
    popularNetwork: {
      flexDirection: 'row',
      alignItems: 'center',
      justifyContent: 'space-between',
      marginVertical: 12,
    },
    button: {
      flex: 1,
    },
    disabledButton: {
      backgroundColor: colors.primary.muted,
    },
    cancel: {
      marginRight: 16,
    },
    blueText: {
      color: colors.primary.default,
      marginTop: 1,
    },
    bottomSection: {
      flex: 1,
      flexDirection: 'column',
    },
    rpcTitleWrapper: {
      flexDirection: 'row',
      alignItems: 'center',
      gap: 4,
    },
  });
