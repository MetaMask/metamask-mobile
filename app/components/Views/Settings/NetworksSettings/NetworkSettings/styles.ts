import { StyleSheet } from 'react-native';

const createStyles = () =>
  StyleSheet.create({
    popularNetwork: {
      flexDirection: 'row',
      alignItems: 'center',
      justifyContent: 'space-between',
      marginVertical: 12,
    },
    popularNetworkImage: {
      width: 20,
      height: 20,
      marginRight: 10,
      borderRadius: 10,
    },
    popularWrapper: {
      flexDirection: 'row',
      alignItems: 'center',
    },
    icon: {
      marginRight: 16,
      marginTop: 4,
    },
  });

export default createStyles;
