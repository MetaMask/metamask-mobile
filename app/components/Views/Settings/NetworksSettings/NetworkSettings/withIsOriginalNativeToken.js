import React, { useEffect, useState } from 'react';
import axios from 'axios';

const CHAIN_ID_NETWORK_URL = 'https://chainid.network/chains.json';

const withIsOriginalNativeToken = (WrappedComponent) => {
  // This is the functional component wrapper that can use hooks
  const WithIsOriginalNativeTokenWrapper = (props) => {
    // Use the useSelector hook to access Redux state
    const [matchedChainNetwork, setMatchedChainNetwork] = useState(null);

    useEffect(() => {
      axios.get(CHAIN_ID_NETWORK_URL).then(({ data: safeChainsList }) => {
        setMatchedChainNetwork({
          safeChainsList: [...safeChainsList],
        });
      });
    }, []);

    // Pass the value from useSelector as a prop to the WrappedComponent
    return (
      <WrappedComponent {...props} matchedChainNetwork={matchedChainNetwork} />
    );
  };

  return WithIsOriginalNativeTokenWrapper;
};

export default withIsOriginalNativeToken;
