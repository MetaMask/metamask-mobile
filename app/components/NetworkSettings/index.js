import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { StyleSheet, View } from 'react-native';
import { colors, fontStyles } from '../../styles/common';
import Engine from '../../core/Engine';
import SettingsList from 'react-native-settings-list'; // eslint-disable-line import/default
import { strings } from '../../../locales/i18n';
import Icon from 'react-native-vector-icons/MaterialIcons';

const styles = StyleSheet.create({
	wrapper: {
		backgroundColor: colors.slate,
		flex: 1
	},
	separator: {
		marginTop: 15
	},
	iconRemove: {
		height: 30,
		marginLeft: 10,
		alignSelf: 'center'
	}
});

/**
 * View that contains all the different
 * network settings
 */
class NetworkSettings extends Component {
	static navigationOptions = () => ({
		title: strings('network.title'),
		headerTitleStyle: {
			fontSize: 20,
			...fontStyles.normal
		}
	});

	static propTypes = {
		/**
		 * Object that contains the whole background state
		 */
		provider: PropTypes.object,
		/**
		 * A list of custom RPCs to provide the user
		 */
		frequentRpcList: PropTypes.array
	};

	changeNetwork = type => {
		const { NetworkController } = Engine.context;
		NetworkController.setProviderType(type);
	};

	mainnet = () => {
		this.changeNetwork('mainnet');
	};

	rinkeby = () => {
		this.changeNetwork('rinkeby');
	};

	ropsten = () => {
		this.changeNetwork('ropsten');
	};

	setRpcTarget = rpcTarget => {
		const { NetworkController } = Engine.context;
		NetworkController.setRpcTarget(rpcTarget);
	};

	removeRpcTarget = rpcTarget => {
		const { PreferencesController } = Engine.context;
		PreferencesController.removeFromFrequentRpcList(rpcTarget);
	};

	render() {
		const { provider, frequentRpcList } = this.props;

		return (
			<View style={styles.wrapper} testID={'network-settings-screen'}>
				<SettingsList borderColor={colors.borderColor} defaultItemSize={50}>
					<SettingsList.Header headerStyle={styles.separator} />
					<SettingsList.Item
						title={'mainnet'}
						titleInfo={provider.type === 'mainnet' ? strings('network.selected') : null}
						onPress={this.mainnet}
						hasNavArrow={false}
					/>
					<SettingsList.Item
						title={'ropsten'}
						titleInfo={provider.type === 'ropsten' ? strings('network.selected') : null}
						onPress={this.ropsten}
						hasNavArrow={false}
					/>
					<SettingsList.Item
						title={'rinkeby'}
						titleInfo={provider.type === 'rinkeby' ? strings('network.selected') : null}
						onPress={this.rinkeby}
						hasNavArrow={false}
					/>
					{frequentRpcList.map(url => {
						const rpcUrlSelected = provider.type === 'rpc' && provider.rpcTarget === url;
						return (
							<SettingsList.Item
								icon={
									rpcUrlSelected ? null : (
										<View style={styles.iconRemove}>
											<Icon
												name="remove-circle"
												size={25}
												color={colors.fontTertiary}
												onPress={() => this.removeRpcTarget(url)} // eslint-disable-line
											/>
										</View>
									)
								}
								title={url}
								titleInfo={rpcUrlSelected ? strings('network.selected') : null}
								key={url}
								onPress={() => this.setRpcTarget(url)} // eslint-disable-line
								hasNavArrow={false}
							/>
						);
					})}
				</SettingsList>
			</View>
		);
	}
}

const mapStateToProps = state => ({
	frequentRpcList: state.backgroundState.PreferencesController.frequentRpcList,
	provider: state.backgroundState.NetworkController.provider
});

export default connect(mapStateToProps)(NetworkSettings);
