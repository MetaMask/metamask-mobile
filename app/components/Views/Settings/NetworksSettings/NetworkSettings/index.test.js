import React from 'react';
import { shallow } from 'enzyme';
import NetworkSettings from './';
import configureMockStore from 'redux-mock-store';
describe('NetworkSettings', () => {
	const mockStore = configureMockStore();
	it('should render correctly', () => {
		const initialState = {
			engine: {
				backgroundState: {
					PreferencesController: {
						frequentRpcList: []
					}
				}
			}
		};
		const wrapper = shallow(
			<NetworkSettings
				navigation={{
					state: { params: {} },
					getParam: () => {
						'network';
					}
				}}
			/>,
			{
				context: { store: mockStore(initialState) }
			}
		);
		expect(wrapper.dive()).toMatchSnapshot();
	});
});
