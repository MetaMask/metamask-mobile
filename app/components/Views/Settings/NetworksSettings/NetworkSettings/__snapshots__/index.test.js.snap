// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NetworkSettings should render correctly 1`] = `
<ForwardRef(SafeAreaView)
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "flex": 1,
      "flexDirection": "column",
    }
  }
  testID="new-rpc-screen"
>
  <KeyboardAwareScrollViewMock
    enableAutomaticScroll={true}
    enableOnAndroid={false}
    enableResetScrollToCoords={true}
    extraHeight={75}
    extraScrollHeight={0}
    keyboardOpeningTime={250}
    style={
      Object {
        "flex": 1,
        "paddingHorizontal": 24,
      }
    }
    viewIsInsideTabBar={false}
  >
    <View
      style={
        Object {
          "flex": 1,
          "paddingVertical": 12,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#000000",
            "fontFamily": "EuclidCircularB-Bold",
            "fontSize": 20,
            "fontWeight": "600",
            "paddingVertical": 12,
          }
        }
        testID="rpc-screen-title"
      >
        New RPC Network
      </Text>
      <Text
        style={
          Object {
            "color": "#000000",
            "fontFamily": "EuclidCircularB-Regular",
            "fontSize": 14,
            "fontWeight": "400",
          }
        }
      >
        Use a custom RPC-capable network via URL instead of one of the provided networks.
      </Text>
      <Text
        style={
          Object {
            "color": "#000000",
            "fontFamily": "EuclidCircularB-Bold",
            "fontSize": 14,
            "fontWeight": "600",
            "paddingVertical": 12,
          }
        }
      >
        Network Name
      </Text>
      <TextInput
        autoCapitalize="none"
        autoCorrect={false}
        onChangeText={[Function]}
        onSubmitEditing={[Function]}
        placeholder="Network Name (optional)"
        placeholderTextColor="#d6d9dc"
        style={
          Array [
            Object {
              "borderColor": "#bbc0c5",
              "borderRadius": 5,
              "borderWidth": 2,
              "fontFamily": "EuclidCircularB-Regular",
              "fontWeight": "400",
              "padding": 10,
            },
            Object {
              "width": "99%",
            },
          ]
        }
        testID="input-network-name"
      />
      <Text
        style={
          Object {
            "color": "#000000",
            "fontFamily": "EuclidCircularB-Bold",
            "fontSize": 14,
            "fontWeight": "600",
            "paddingVertical": 12,
          }
        }
      >
        RPC Url
      </Text>
      <TextInput
        autoCapitalize="none"
        autoCorrect={false}
        onBlur={[Function]}
        onChangeText={[Function]}
        onSubmitEditing={[Function]}
        placeholder="New RPC Network"
        placeholderTextColor="#d6d9dc"
        style={
          Array [
            Object {
              "borderColor": "#bbc0c5",
              "borderRadius": 5,
              "borderWidth": 2,
              "fontFamily": "EuclidCircularB-Regular",
              "fontWeight": "400",
              "padding": 10,
            },
            Object {
              "width": "99%",
            },
          ]
        }
        testID="input-rpc-url"
      />
      <Text
        style={
          Object {
            "color": "#000000",
            "fontFamily": "EuclidCircularB-Bold",
            "fontSize": 14,
            "fontWeight": "600",
            "paddingVertical": 12,
          }
        }
      >
        Chain ID
      </Text>
      <TextInput
        autoCapitalize="none"
        autoCorrect={false}
        keyboardType="numbers-and-punctuation"
        onBlur={[Function]}
        onChangeText={[Function]}
        onSubmitEditing={[Function]}
        placeholder="Chain ID"
        placeholderTextColor="#d6d9dc"
        style={
          Array [
            Object {
              "borderColor": "#bbc0c5",
              "borderRadius": 5,
              "borderWidth": 2,
              "fontFamily": "EuclidCircularB-Regular",
              "fontWeight": "400",
              "padding": 10,
            },
            Object {
              "width": "99%",
            },
          ]
        }
        testID="input-chain-id"
      />
      <Text
        style={
          Object {
            "color": "#000000",
            "fontFamily": "EuclidCircularB-Bold",
            "fontSize": 14,
            "fontWeight": "600",
            "paddingVertical": 12,
          }
        }
      >
        Symbol
      </Text>
      <TextInput
        autoCapitalize="none"
        autoCorrect={false}
        onChangeText={[Function]}
        onSubmitEditing={[Function]}
        placeholder="Symbol (optional)"
        placeholderTextColor="#d6d9dc"
        style={
          Array [
            Object {
              "borderColor": "#bbc0c5",
              "borderRadius": 5,
              "borderWidth": 2,
              "fontFamily": "EuclidCircularB-Regular",
              "fontWeight": "400",
              "padding": 10,
            },
            Object {
              "width": "99%",
            },
          ]
        }
        testID="input-network-symbol"
      />
      <Text
        style={
          Object {
            "color": "#000000",
            "fontFamily": "EuclidCircularB-Bold",
            "fontSize": 14,
            "fontWeight": "600",
            "paddingVertical": 12,
          }
        }
      >
        Block Explorer URL
      </Text>
      <TextInput
        autoCapitalize="none"
        autoCorrect={false}
        onChangeText={[Function]}
        onSubmitEditing={[Function]}
        placeholder="Block Explorer URL (optional)"
        placeholderTextColor="#d6d9dc"
        style={
          Array [
            Object {
              "borderColor": "#bbc0c5",
              "borderRadius": 5,
              "borderWidth": 2,
              "fontFamily": "EuclidCircularB-Regular",
              "fontWeight": "400",
              "padding": 10,
            },
            Object {
              "width": "99%",
            },
          ]
        }
      />
    </View>
    <View
      style={
        Object {
          "alignSelf": "flex-end",
          "flexDirection": "row",
          "marginVertical": 12,
        }
      }
    >
      <View
        style={
          Object {
            "alignSelf": "flex-end",
            "flex": 1,
            "flexDirection": "column",
          }
        }
      >
        <StyledButton
          disabled={true}
          disabledContainerStyle={
            Object {
              "opacity": 0.6,
            }
          }
          onPress={[Function]}
          styleDisabled={
            Object {
              "opacity": 0.6,
            }
          }
          testID="network-add-button"
          type="confirm"
        >
          Add
        </StyledButton>
      </View>
    </View>
  </KeyboardAwareScrollViewMock>
</ForwardRef(SafeAreaView)>
`;
